crates/cli/src/opts/build/revive.rs

```diff
@@ -0,0 +1,60 @@
+use clap::Parser;
+use foundry_config::revive::ReviveConfig;
+use serde::Serialize;
+use std::path::PathBuf;
+
+#[derive(Clone, Debug, Default, Serialize, Parser)]
+#[clap(next_help_heading = "Revive configuration")]
+/// Compiler options for revive
+/// TODO: We need to add more revive specific arguments
+pub struct ReviveOpts {
+    #[clap(
+        value_name = "REVIVE_COMPILE",
+        help = "Enable compiling with revive",
+        long = "revive-compile",
+        action = clap::ArgAction::SetTrue,
+        default_value_t = false
+    )]
+    pub revive_compile: bool,
+
+    #[clap(help = "Solc path to be used by revive", long = "solc-path", value_name = "SOLC_PATH")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub solc_path: Option<PathBuf>,
+
+    #[clap(
+        long = "revive-path",
+        visible_alias = "revive",
+        help = "Specify a revive path to be used",
+        value_name = "REVIVE_PATH"
+    )]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub revive_path: Option<PathBuf>,
```

Comment:
I see that in compilers crate we have: `Resolc::new(which::which("resolc").unwrap(), solc).unwrap());`
So how is it working?

Reply:
This is for when the user initialises Resolc/Revive usign `Resolc::default()` the regular flow uses the path via `Resolc::new(revive_path,solc_compiler)` theres 3 ways to initialise resolc (default),revive_path +SolcCompiler and revive_path,+solc_path